File: csharp/2016/pride-and-prejudice-and-zombies-add.cs
using System;
using System.Threading.Tasks;
using Amazon.DynamoDBv2;
using Amazon.DynamoDBv2.Model;
using System.Collections.Generic;

namespace MovieDatabase
{
    class Program
    {
        static async Task Main(string[] args)
        {
            // Create a MovieRepository instance
            var movies = new MovieRepository();

            // Add "Pride and Prejudice and Zombies" to the database
            // This demonstrates how to insert a new item into DynamoDB
            await movies.InsertAsync(
                title: "Pride and Prejudice and Zombies",
                year: 2016,
                plot: "Five sisters in 19th century England must cope with the pressures to marry while protecting themselves from a growing population of zombies.",
                rating: 5.8
            );

            // Confirm that the movie was added by retrieving it
            var movie = await movies.SelectAsync(
                title: "Pride and Prejudice and Zombies",
                year: 2016
            );

            if (movie != null)
            {
                // The movie was found
                Console.WriteLine($"Movie found: {movie}");
            }
            else
            {
                // The movie was not found
                Console.WriteLine("Movie not found");
            }
        }
    }
}

File: python/2016/pride-and-prejudice-and-zombies-add.py

from movies import MovieRepository

# Create a new movie repository using the default configuration
movies = MovieRepository()

# Add Pride and Prejudice and Zombies to the database
movies.insert(
    title="Pride and Prejudice and Zombies", 
    year=2016, 
    plot="Five sisters in 19th century England must cope with the pressures to marry while protecting themselves from a growing population of zombies.", 
    rating=5.8
)

# Confirm that the movie was added
movie = movies.select(
    title="Pride and Prejudice and Zombies", 
    year=2016
)
if movie:
    # The movie was found
    print(f"Movie found: {movie}")
else:
    # The movie was not found
    print("Movie not found")
    